RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/thread.cpp,v
retrieving revision 1.82
diff -r1.82 thread.cpp
487c487,489
<     if ( thread->m_internal->GetState() == STATE_EXITED )
---
>     bool isExited = thread->m_internal->GetState() == STATE_EXITED;
> 
>     if ( isExited )
501a504,506
>     }
> 
>     thread->OnExit();
502a508,512
>     // save IsDetached as thead object can be deleted 'JOINABLE' threads after
>     // state is changed to STATE_EXITED
>     bool isDetached = thread->IsDetached();
> 
>     if(!isExited) {
505d514
< 
509,510d517
<     thread->OnExit();
< 
512c519
<     thread->m_internal->LetDie();
---
>     if(isDetached) thread->m_internal->LetDie();

 	  	 
